### PR TITLE
Use fixed sized array for fileset.IDs

### PR DESCRIPTION
**Pull request by brendoncarroll**

### CODE REVIEW

File: src/internal/storage/fileset/compaction.go [hunk 1]
```diff
@@ -16,7 +16,11 @@ func (s *Storage) IsCompacted(ctx context.Context, id ID) (bool, error) {
 	}
 	switch x := md.Value.(type) {
 	case *Metadata_Composite:
-		ids, err := s.Flatten(ctx, stringsToIDs(x.Composite.Layers))
+		ids, err := stringsToIDs(x.Composite.Layers)
```
Comment (brycemcanally) [inline src/internal/storage/fileset/compaction.go:19]: Shouldn't this be `PointsTo`?

File: src/internal/storage/fileset/fileset.go [hunk 2]
```diff
@@ -1,42 +1,73 @@
...
+	id := ID{}
+	if _, err := rand.Read(id[:]); err != nil {
+		panic(err)
+	}
+	return id
 }
 
 // ParseID parses a string into an ID or returns an error
 func ParseID(x string) (*ID, error) {
 	if len(x) < 32 {
```
Comment (brycemcanally) [inline src/internal/storage/fileset/fileset.go:29]: Should we also check if the id is too long?

File: src/internal/storage/fileset/fileset.go [hunk 3]
```diff
@@ -1,42 +1,73 @@
...
 
+// Scan implements sql.Scanner
+func (id *ID) Scan(src interface{}) error {
+	x, ok := src.([]byte)
+	if !ok {
+		return errors.Errorf("scanning fileset.ID: can't turn %T into fileset.ID", src)
+	}
+	x = bytes.Replace(x, []byte{'-'}, []byte{}, -1)
+	*id = ID{}
+	if len(x) < 32 {
```
Comment (brycemcanally) [inline src/internal/storage/fileset/fileset.go:59]: Maybe this should be a helper function that we can call to validate a hex encoded ID?

Comment (brendoncarroll) [reply src/internal/storage/fileset/fileset.go:59]: what other place would we use that?

Comment (brycemcanally) [reply src/internal/storage/fileset/fileset.go:59]: Where I commented above. It seems like the validation should check for an exact length and be implemented in one place through a helper function.

**Review by brycemcanally — APPROVED** [review]
lgtm

**Pull request merged**